### ISSUE TITLE
Unable to Export Reusable Blocks

### ISSUE THREAD

**Issue by macmanx2** [Automattic/wp-calypso#39153]
#### Steps to reproduce
1. On any non-AT WordPress.com site, create a Reusable Block: https://en.support.wordpress.com/wordpress-editor/blocks/reusable-block/
2. Visit `/wp-admin/edit.php?post_type=wp_block` on that site.
3. Attempt to choose "Export as JSON" for any block.
4. Notice that nothing happens.

#### What I expected
Under WordPress 5.3.2, the browser's download prompt appears.

#### What happened instead
Under any non-AT WordPress.com site, nothing happens.

#### Browser / OS version
macOS 10.15.3, reproduced under Firefox 72.0.2, Safari 13.0.5, and Chrome 79.0.3945.130.

#### Screenshot / Video
![Screen Shot 2020-01-29 at 11 09 18](https://user-images.githubusercontent.com/4856235/73392153-20b8e980-428e-11ea-98fd-a9bf870a8cef.png)

#### Context / Source
Reported via https://en.forums.wordpress.com/topic/in-the-reusable-blocks-editor/

**Comment by kriskarkoski** [comment]
Confirmed I can still reproduce this issue on a Premium site in Chrome 79.0.3945.13

**Comment by cathymcbride** [comment]
Still reproing as of today. Adding a screenshot of the console and moving onto another board.
![image](https://user-images.githubusercontent.com/16654218/94301522-b2214e00-ff38-11ea-8f31-9451f52e90a9.png)

**Comment by cathymcbride** [comment]
@sirreal if doesn't fall within your teams purview, let me know!

**Comment by sirreal** [comment]
I took a quick look at this. I've added it to the create maintenance board because it's a bug and not strictly related to editor integration.

@macmanx2 or @kriskarkoski do you have an idea of how big the impact is here? As far as I can tell you must manually introduce this URL to reach the page. Are users encountering this issue? Do you know how the page can be reached?

This is caused by an issue with `public-api`. There is a request triggered to `https://public-api.wordpress.com/wp/v2/types/wp_block?_locale=user` that fails with the following error:

> Access to fetch at 'https://public-api.wordpress.com/wp/v2/types/wp_block?_locale=user' from origin 'https://example.wordpress.com' has been blocked by CORS policy: Request header field x-wp-nonce is not allowed by Access-Control-Allow-Headers in preflight response.

There are a few problems with the request:

- The URL should include the site fragment. `https://public-api.wordpress.com/wp/v2/types/wp_block?_locale=user` should be `https://public-api.wordpress.com/wp/v2/sites/[[ SITE_ID ]]/types/wp_block?_locale=user`.
- The `x-wp-nonce` header either should not be provided (it doesn't seem necessary) or it should be allowed on these requests by `Access-Control-Allow-Headers`.

**Comment by kriskarkoski** [comment]
Given the lack of addition reports I'd say the impact is minimal (I tested as part of a bug sweep for HACK week) since no additional reports have come in here in almost 9 months.

**Comment by cathymcbride** [comment]
Thank you @kriskarkoski Do you know if you can get to this page via the UI?

**Comment by kriskarkoski** [comment]
@cathymcbride I do not know of way to access it in the UI. I was thinking there was a blog RC option, but it seems that "Don't redirect WP-Admin importers to Calypso" only applies to imports, not exports.

**Comment by cathymcbride** [comment]
OK - in that case I'm going to leave it at low pri for the moment, since it's quite hard to get to.

**Comment by kriskarkoski** [comment]
Cool, I agree with it being low priority, possibly even wontfix if it works or is hidden from being accessible from the UI

**Comment by andrewserong** [comment]
I think it's fairly unlikely that users will get there this way, but to get there via the UI, you can:

1. From Calypso on your site click the link to go to `wp-admin`.
2. Go to any post
3. Create a re-usable block, then from the left hand inserter, select the Reusable tab, then click on Manage all reusable blocks:

![image](https://user-images.githubusercontent.com/14988353/100028581-66f5bf80-2e43-11eb-8a22-53de73c2e06f.png)

I agree it's low-priority though, so I'll place it at the bottom of the prioritised backlog for now.

**Comment by TeniCola** [comment]
Hi team, I had another user who _did_ manage to find this option in their site -- we're following up in 4021146-zen. In the preceding livechat when we asked why they wanted this feature, they explained:

> 1. Keep a backup of the blocks on my computer
> 2. Import them to another site

I will let them know the bug has been reported, and that they can save the blocks from the code editor in the meantime. But if we don't intend to fix this, it may be worth making the call and removing the link entirely; it'll be hard to explain to this user that this link "doesn't work" when it's currently there for them to discover and click.

**Comment by kspilarski** [comment]
Same case in 4098102-zen, I was able to replicate the issue as well. Nothing happens when "Export as JSON" is clicked under a reusable block.

**Comment by Jaccuse** [comment]
Another user ran into this in 36176730-hc

**Comment by mriyazuddin** [comment]
- [ ] Another one reported here: #5442039-zen

**Comment by hngdny** [comment]
Also reported here 5637820-zd-woothemes

**Comment by suyogyashukla** [comment]
Another report in 8213552-zd-a8c

**Comment by liviopv** [comment]
This has been working in Gutenberg for a while now. Closing.

**Comment by gamebits** [comment]
@liviopv I just tried exporting a reusable block as JSON from +difmp2 and experienced the issue that @macmanx2 originally reported.

Is wpcom running an older version of Gutenberg where this issue is still occurring?

**Comment by liviopv** [comment]
@gamebits 

> a reusable block as JSON from +difmp2

P2 runs on outdated GB versions. Right now, it is on GB 18. And because it's a non-standard theme/setup, it's not a guarantee that certain features will be stable. If you can only repro in P2, can you report it to https://github.com/Automattic/p2/issues ?

Here's how Exporting / Importing patterns should work as of GB 18.5:


https://github.com/Automattic/wp-calypso/assets/14153300/799e4b58-aa42-4659-89e8-796bc9cf2450
